statsbestcustomers.php

```diff
@@ -167,12 +171,12 @@ public function getData()
 				AND o.valid
 			), 0) as totalValidOrders
 		FROM `' . _DB_PREFIX_ . 'customer` c
-		INNER JOIN `' . _DB_PREFIX_ . 'orders` o ON o.id_customer = c.id_customer
 		LEFT JOIN `' . _DB_PREFIX_ . 'guest` g ON c.`id_customer` = g.`id_customer`
 		LEFT JOIN `' . _DB_PREFIX_ . 'connections` co ON g.`id_guest` = co.`id_guest` AND co.date_add BETWEEN ' . $this->getDate() . '
-		WHERE o.date_add BETWEEN ' . $this->getDate()
-            . Shop::addSqlRestriction(Shop::SHARE_CUSTOMER, 'c') .
-            ' GROUP BY c.`id_customer`, c.`lastname`, c.`firstname`, c.`email`';
+		WHERE co.date_add BETWEEN ' . $this->getDate() . '
+		AND c.id_customer IN (SELECT o.id_customer FROM `' . _DB_PREFIX_ . 'orders` o)
```

Comment:
Why using a subquery when you can use an INNER join? :thinking:

Reply:
INNER JOIN duplicate lines for each order, so the count of total connections are invalid.

Reply:
You can also use distinct :thinking: 
I'm afraid of performance problem when you have thousands of orders..